### ISSUE TITLE
Cloudflare Bypassing

### ISSUE THREAD

**Issue by dulleh** [dulleh/akhyou#28]
This comes down to cookies not being recognized as invalid (for one reason or another) afaik.

The current code alleviates this through force-solving Cloudflare but if the cookie does work this results in null pointers as elements expected to be on the Cloudflare loading screen aren't on a regular page.

```
public Response execute(Request request) throws IOException, CloudflareException {
        Response resp = client.newCall(request).execute();
        String refresh = resp.header("Refresh");
        String server = resp.header("Server");

        List<HttpCookie> cookies = cookieManager.getCookieStore().get(request.uri());
        boolean hasCookie = Stream.of(cookies).anyMatch(c -> c.getName().equals("cf_clearance"));

        if (hasCookie)
            return resp;
        //doesn't have cookie
        if (forceSolve) {
            return solveCloudflare(resp);
        } else if (refresh != null && refresh.contains("URL=/cdn-cgi/") &&
                server != null && server.equals("cloudflare-nginx")) {

            //System.out.println("solving cloudflare");
            return solveCloudflare(resp);
        }

        return resp;
    }
```

So I'm suggesting that the error lies in:

```
        else if (refresh != null && refresh.contains("URL=/cdn-cgi/") &&
                server != null && server.equals("cloudflare-nginx"))
```

Perhaps changing the "and" requirement to "or" would work? I'll test this later. 

I'm also updating to Okhttp3 and removing the AsyncTask in [CloudflareHttpClient](https://github.com/dulleh/akhyou/blob/master/akhyou2/src/main/java/dulleh/akhyou/Utils/CloudflareHttpClient.java) to RxJava which may affect how handling this error pans out with the new CookieJar interface of Okhttp3.

**Comment by dulleh** [comment]
https://github.com/dulleh/akhyou/issues/27#issuecomment-184724020

**Comment by dulleh** [comment]
Should be fixed as of 0e62499ed3acdfad990a9410db701704b0f658c9

**Comment by dulleh** [comment]
^ At the risk of presenting new bugs.

**Comment by dulleh** [comment]
Fixed as of 292c9ac825ec00cfeb8f4ee3680d010921169b21
